nucleus/deploy/client.py

```diff
@@ -0,0 +1,286 @@
+        """
+        Initializes a Scale Deploy Client.
+
+        Parameters:
+            api_key: Your Scale API key
+            endpoint: The Scale Deploy Endpoint (this should not need to be changed)
+        """
+        self.connection = Connection(api_key, endpoint)
+
+    def __repr__(self):
+        return f"DeployClient(connection='{self.connection}')"
+
+    def __eq__(self, other):
+        return self.connection == other.connection
+
+    def create_model_bundle(
+        self,
+        model_bundle_name: str,
+        load_predict_fn: Callable[[DeployModel_T], Callable[[Any], Any]],
+        model: Optional[DeployModel_T] = None,
+        load_model_fn: Optional[Callable[[], DeployModel_T]] = None,
+    ) -> ModelBundle:
+        """
+        Grabs a s3 signed url and uploads a model bundle to Scale Deploy.
+        A model bundle consists of a "model" and a "load_predict_fn", such that
+        load_predict_fn(model) returns a function predict_fn that takes in model input and returns model output.
+        Pre/post-processing code can be included inside load_predict_fn/model.
+
+        Parameters:
+            model_bundle_name: Name of model bundle you want to create. This acts as a unique identifier.
+            model: Typically a trained Neural Network, e.g. a Pytorch module
+            load_model_fn: Function that when run, loads a model, e.g. a Pytorch module
+            load_predict_fn: Function that when called with model, returns a function that carries out inference
+        """
+
+        if (model is not None and load_model_fn is not None) or (
+            model is None and load_model_fn is None
+        ):
+            raise ValueError(
+                "Exactly one of model and load_model_fn should be non-None"
+            )
+
+        # Grab a signed url to make upload to
+        model_bundle_s3_url = self.connection.post({}, "model_bundle_upload")
+        if "signedUrl" not in model_bundle_s3_url:
+            raise Exception(
+                "Error in server request, no signedURL found"
+            )  # TODO code style broad exception
+        s3_path = model_bundle_s3_url["signedUrl"]
+        raw_s3_url = f"s3://{model_bundle_s3_url['bucket']}/{model_bundle_s3_url['key']}"
+
+        # Make bundle upload
+        if model is not None:
+            bundle = dict(model=model, load_predict_fn=load_predict_fn)
+        else:
+            bundle = dict(
+                load_model_fn=load_model_fn, load_predict_fn=load_predict_fn
+            )
+        serialized_bundle = cloudpickle.dumps(bundle)
+        requests.put(s3_path, data=serialized_bundle)
+
+        self.connection.post(
+            payload=dict(id=model_bundle_name, location=raw_s3_url),
+            route="model_bundle",
+        )  # TODO use return value somehow
+        # resp["data"]["bundle_name"] should equal model_bundle_name
+        # TODO check that a model bundle was created and no name collisions happened
```

Comment:
The connection class already raises an error for a non-200 status code: https://github.com/scaleapi/nucleus-python-client/blob/b18c412367e2f8ff8a7ba24e1cdb25b11b204b1b/nucleus/connection.py#L73

I would advise returning non-200 on name collision